### PR TITLE
max-step after load checkpoint

### LINKED ISSUE #1922: Trainer max_steps does not work when reloaded from checkpoint (if value is already exceeded)
## 🐛 Bug 

The trainer flag for `max_steps` is not respected when you reload the Trainer from a checkpoint.

### To Reproduce

This works:
```
import tests.base.utils as tutils
from pytorch_lightning import Trainer, LightningModule
from pytorch_lightning.callbacks import EarlyStopping, LearningRateLogger, ModelCheckpoint
from tests.base import EvalModelTemplate

def test_trainer_max_steps():
    model = EvalModelTemplate()
    trainer = Trainer(max_steps=1)
    trainer.fit(model)
    assert trainer.global_step == 1
```

This breaks:
```
def test_trainer_exceed_max_steps_from_checkpoint():
    model = EvalModelTemplate()
    checkpoint_callback = ModelCheckpoint(save_top_k=1)
    trainer = Trainer(checkpoint_callback=checkpoint_callback, max_epochs=4)
    trainer.fit(model)
    step_count = trainer.global_step

    checkpoint_filepath = checkpoint_callback.kth_best_model
    trainer = Trainer(max_steps=1,
                    resume_from_checkpoint=checkpoint_filepath)
    trainer.fit(model)
    # current step exceeds max_steps, training should stop immediately
    assert trainer.global_step == step_count
```

### Expected behavior

Both of the above tests that I've written should pass.

We should also probably warn the user when they pass in a `max_steps` or `max_epochs` which exceeds the current state when reloading from a checkpoint.

### Environment

* CUDA:
        - GPU:
        - available:         False
        - version:           None
* Packages:
        - numpy:             1.18.1
        - pyTorch_debug:     False
        - pyTorch_version:   1.4.0
        - pytorch-lightning: 0.7.7-dev
        - tensorboard:       2.1.0
        - tqdm:              4.42.0
* System:
        - OS:                Darwin
        - architecture:
                - 64bit
                - 
        - processor:         i386
        - python:            3.7.6
        - version:           Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Fixes #1922

**TODO:**
- [x] Add test

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
it seems that some other test is failing regarding LR
cc: @SkafteNicki

**Comment by SkafteNicki** [comment]
@Borda it probably has something to do with these lines, that tells the lr finder to stop the search early if the loss is diverging too much:
https://github.com/PyTorchLightning/pytorch-lightning/blob/bd49b07fbba09b1e7d8851ee5a1ffce3d5925e9e/pytorch_lightning/trainer/lr_finder.py#L389-L393
With this PR, it seems to stop too early which results in the `suggestion` algorithm failing because not enough points have been logged to calculate gradient (which in turn make the optimizer initialization fail because the learning rate is set to `None`)

File: pytorch_lightning/trainer/training_loop.py [hunk 1]
```diff
@@ -524,7 +524,7 @@ def run_training_epoch(self):
             self.increment_accumulated_grad_global_step()
 
             # max steps reached, end training
-            if self.max_steps is not None and self.max_steps == self.global_step:
+            if self.max_steps is not None and self.max_steps >= self.global_step:
```
Comment (awaelchli) [inline pytorch_lightning/trainer/training_loop.py:527]:
```suggestion
            if self.max_steps is not None and self.max_steps <= self.global_step:
```
pretty sure this is what you mean, since global step is the variable that increases and max_steps is the one that is constant. 
Besides that, for an easier understanding, i would just swap the variables around

**Comment by awaelchli** [comment]
I think this deserves the "allowed before v1.0"

**Comment by Borda** [comment]
@PyTorchLightning/core-contributors it seems that it is not the right way, the case is about resetting epoch/step counter when a new training starts, but how to distinguish between loading checkout as resume training and loading old model for restraining...

**Comment by awaelchli** [comment]
I think resetting the step counters is not the right way. when loading a model, it should be known how long it was trained before. resetting that information is dangerous as you will lose part of reproducibility. If max_steps was reached, and you want to load and continue training, I see the only right way is to increase max_steps or max_epochs.

File: pytorch_lightning/trainer/training_loop.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/trainer/training_loop.py:527]: nonetheless, I think this fix in my suggestion is correct even if it does not fix what you orignially had in mind.
